### DISCUSSION TITLE
Jog command from MEGA/RAMPS/Reprap Controller not working

### DISCUSSION THREAD

**Discussion by mvackel** [Paciente8159/uCNC#533] [Config and issues] [answered]
I'm using the new 1.8.0 uCNC with Mega/RAMPS/RepRap smart display controller.
The Jog commands issued from the controller is not working anymore. Sometimes when the encoder is turned many times very quickly the jog is sent to the machine, but most of the times its unresponsive.

**Comment by Paciente8159** [comment]
I will run more tests on this.
Maybe the extension modules has some bad implementation of the multi-stream.

Just a few notes.

- Multi-stream responds to all realtime commands from all sources almost immediately.
- Regular commands (like JOG) are answered by priority order. The priority is defined by the order the streams were added. UARTs have the highest priority, then USB then WIFI, then Bluetooth, then the remaining sources in the respective order.

For example if you are running commands from the UART the display commands are waiting for the channel to become free to transmit.

On that note. If don't connect the board to any program does the display improves the response?

**Comment by Paciente8159** [comment]
Ok confirmed there is an issue. I will review the code and give you feedback. I will open an issue.

**Comment by Paciente8159** [comment]
This is still being tested. This is a hard bug to track due to the lack of debugging tools on AVR.
I think the issue is something related to a loop reentrancy or reentrancy protection.
The unlock  command uses the same stream buffer mechanism that is dedicated to the graphic display and works ok. I'm testing different solutions.

I might need to implement a simulation on a different board with a different display since this one does not seem to work with 3.3v.

**Comment by Paciente8159** [comment]
I've tracked the issue and I'm working on the patches to fix this. As soon as it's sorted I'll give feedback.

**Comment by Paciente8159** [comment, marked as answer]
I've implemented the changes to the core code and graphic display module, available at the master branch.

Took quite some logic refactoring, so I hope I did not introduced any new issue.
Please report if you find anything wrong.
Thanks.
